### ISSUE TITLE
API Connectivity Errors

### ISSUE THREAD

**Issue by Horse2370** [hjdhjd/homebridge-ratgdo#10]
### Homebridge Ratgdo

v2.2.0

### Homebridge Platform and OS

linux x86_64 Ubuntu Core 22 and Docker 24.0.5 (API: 1.43) 

### Homebridge Version

v1.7.0

### Node Version

v20.12.2

### Ratgdo ESPHome Firmware Version

2024.5.0

### Describe the problem

I have two ratgdo 2.5i's connected to LiftMaster 8500W which are Security 2.0+ 
Reflashed from MQTT to make use of your HomeBridge plugin. Flashed using WebFlasher via chrome a couple of days ago.
I do have ESPHome dashboard installed, however have not adopted these ratgdo's to keep this as vanilla as possible.
Everything on the same subnet.
WiFi AP for these is in the Garage, very close.
No WiFi disconnects seen on my UniFi network for either device.
Docker host is 

### Homebridge HBR JSON configuration

```json
I anonymized the domain and the bridge username

{
    "mqttTopic": "Home/1st_Floor/Garage",
    "mqttUrl": "mqtt://mqtt.mydomain.us",
    "name": "Ratgdo",
    "platform": "Ratgdo",
    "_bridge": {
        "username": "0A:0B:0C:0D:0E:0F",
        "port": 54290
    },
    "options": [
        "Enable.Opener.Dimmer"
    ]
}
```

### Relevant log output

```shell
ratgdo web logs have the following errors

17:31:43 | [W] | [component:170] | Component api cleared Warning flag
17:33:43 | [W] | [api.connection:104] | : Connection closed
17:33:43 | [W] | [component:157] | Component api set Warning flag: unspecified

The homebridge plugin logs have the following errors

[5/23/2024, 4:26:04 PM] [Ratgdo] ratgdov2.5i 8e54d7: Ratgdo API error sending command: Connection to the Ratgdo controller has been reset.
[5/23/2024, 4:26:15 PM] [Ratgdo] ratgdov2.5i 8e54d7: Ratgdo API error sending command: Connection to the Ratgdo controller has timed out.
[5/23/2024, 4:26:15 PM] [Ratgdo] ratgdov2.5i 8e54d7: Connection to the Ratgdo controller has timed out while listening for events.
[5/23/2024, 4:26:15 PM] [Ratgdo] ratgdov2.5i 8e54d7: Connection to the Ratgdo controller refused.
[5/23/2024, 4:26:16 PM] [Ratgdo] ratgdov2.5i 8e54d7: Ratgdo API error sending command: Connection to the Ratgdo controller has timed out.
[5/23/2024, 4:29:40 PM] [Ratgdo] ratgdov2.5i 8e54d7: Ratgdo API error sending command: Connection to the Ratgdo controller has been reset.
[5/23/2024, 4:29:41 PM] [Ratgdo] ratgdov2.5i 8e54d7: Connection to the Ratgdo controller has timed out while listening for events.
[5/23/2024, 4:30:11 PM] [Ratgdo] ratgdov2.5i 8e54d7: Ratgdo connected.
```

### Acknowledgment that you are only running non-beta firmware releases from Apple, ESPHome, or Ratgdo

- [X] I acknowledge that I am not running any beta firmwares or operating systems from Apple, ESPHome, or Ratgdo.

**Comment by hjdhjd** [comment]
Read https://github.com/hjdhjd/homebridge-ratgdo/issues/9#event-12900011017

TL;DR: the 2024.5.x firmwares are unfortunately a regression for ESPHome. You will need to install an earlier firmware.

**Comment by Horse2370** [comment]
How can I get the earlier firmware? I used https://ratgdo.github.io/esphome-ratgdo/ to flash directly from web.
what is the location and what image do you recommend? Apologies for such a basic question.

**Comment by hjdhjd** [comment]
Read the link I pointed you to. Literally explains the steps.

**Comment by Horse2370** [comment]
Thanks, I was thinking there was a way to select older firmware from the Web flashing.
recreated the ESP Home container using the 2024.4.2 image and using your custom yaml have flashed both ratgdo's and they do work significantly better. Although every now and then the ratgdo logs show API timeout every second for 60 seconds. WiFi is connected with no issues. but seems to recover and not affect operation.

Thanks for assisting.

**Comment by hjdhjd** [comment]
You might want to reflash and erase the Ratgdos a second time for good measure and see if it clears out the lingering issues. Glad you're up and running, best of luck.
